File: LMIS/LMIS.Web/Modules/ServerTypes/InventoryManage/BookStoreColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BookStoreRow } from "./BookStoreRow";

export interface BookStoreColumns {
    BookStoreId: Column<BookStoreRow>;
    BookName: Column<BookStoreRow>;
    BookshelfLocation: Column<BookStoreRow>;
    Inventory: Column<BookStoreRow>;
    CreateTime: Column<BookStoreRow>;
    UpdateTime: Column<BookStoreRow>;
}

export class BookStoreColumns extends ColumnsBase<BookStoreRow> {
    static readonly columnsKey = 'InventoryManage.BookStore';
    static readonly Fields = fieldsProxy<BookStoreColumns>();
}

File: LMIS/LMIS.Web/Modules/InventoryManage/BookStore/BookStoreColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.InventoryManage.Columns;

[ColumnsScript("InventoryManage.BookStore")]
[BasedOnRow(typeof(BookStoreRow), CheckNames = true)]
public class BookStoreColumns
{
    [AlignRight]
    public long BookStoreId { get; set; }
    public string BookName { get; set; }
    public string BookshelfLocation { get; set; }
    public long Inventory { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}